### PR TITLE
feat(ignore-files): ignore search files by config

### PR DESCRIPTION
**Pull request by wfatec**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@easyops-cn/docusaurus-search-local",
-  "version": "0.16.3",
+  "version": "0.16.4",
```
Comment (weareoutman) [inline package.json:3]: Releases should only be published after merged into master branch (and for repo owners only).

File: src/server/utils/processDocInfos.ts [hunk 2]
```diff
@@ -25,6 +26,12 @@ export function processDocInfos(
         );
       }
       const route = url.substr(baseUrl.length);
+
+      // ignore files
+      if (ignoreFiles?.some((reg: RegExp | string) => route.match(reg))) {
```
Comment (weareoutman) [inline src/server/utils/processDocInfos.ts:31]: I prefer to consider strings as glob match patterns. It's ok to support RegExp only first, but we should not accept strings as implicit RegExps.

**Review by weareoutman — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (wfatec) [reply package.json]: > Releases should only be published after merged into master branch (and for repo owners only).

done~

File: src/server/utils/processDocInfos.ts (hunk 2 above)
Comment (wfatec) [reply src/server/utils/processDocInfos.ts]: Yes, that's right, i have fix this problem~

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,17 +2,21 @@
...
+### [0.16.3](https://www.github.com/easyops-cn/docusaurus-search-local/compare/v0.16.1...v0.16.3) (2021-05-13)
+
+### Bug Fixes
 
+- missing runtime dependency tslib ([2f9d553](https://www.github.com/easyops-cn/docusaurus-search-local/commit/2f9d553b4a3a9596a0f738b80e25049b0a971d77))
 
 ### Miscellaneous Chores
 
-* release 0.16.3 ([b99a906](https://www.github.com/easyops-cn/docusaurus-search-local/commit/b99a9065af9a7dbfc4f52e52eec0935629ae13c5))
+- release 0.16.3 ([b99a906](https://www.github.com/easyops-cn/docusaurus-search-local/commit/b99a9065af9a7dbfc4f52e52eec0935629ae13c5))
```
Comment (weareoutman) [inline CHANGELOG.md:19]: This file should be reverted.

File: src/server/utils/validateOptions.ts [hunk 4]
```diff
@@ -25,6 +31,8 @@ const schema = Joi.object<PluginOptions>({
   highlightSearchTermsOnTargetPage: Joi.boolean().default(false),
   searchResultLimits: Joi.number().default(8),
   searchResultContextMaxLength: Joi.number().default(50),
+  ignoreFiles: isArrayOfStringsOrRegExpsOrStringOrRegExp.default([]),
+  // ignoreFiles: Joi.any(),
```
Comment (weareoutman) [inline src/server/utils/validateOptions.ts:35]: Always remove unnecessary comments.

**Review by weareoutman — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (wfatec) [reply CHANGELOG.md:13]: Done.

File: src/server/utils/validateOptions.ts (hunk 4 above)
Comment (wfatec) [reply src/server/utils/validateOptions.ts]: Done.

**Review by weareoutman — APPROVED** [review]

File: src/server/utils/processDocInfos.ts [hunk 5]
```diff
@@ -25,6 +26,19 @@ export function processDocInfos(
...
+      if (
+        ignoreFiles?.some((reg: RegExp | string) => {
+          if (typeof reg === "string") {
+            return route === reg;
+          }
+          return route.match(reg);
+        })
+      ) {
+        return;
+      }
```
Comment (weareoutman) [inline src/server/utils/processDocInfos.ts:40]: @wfatec Test coverage decreased, and these lines above are not covered. See https://coveralls.io/builds/40073221/source?filename=src%2Fserver%2Futils%2FprocessDocInfos.ts#L33

**Review by weareoutman — CHANGES_REQUESTED** [review]

**Review by weareoutman — APPROVED** [review]

**Comment by weareoutman** [comment]
@wfatec Thanks for your contribution! It will be released via 0.17.0 soon.

**Pull request merged**